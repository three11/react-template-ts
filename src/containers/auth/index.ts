export { default as Login } from './login';
export { default as PasswordReset } from './password-reset';
export { default as Signup } from './signup';
