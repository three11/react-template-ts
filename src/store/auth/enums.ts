export enum AuthActionType {
	LOGIN_REQUEST = 'LOGIN_REQUEST',
	LOGIN_SUCCESS = 'LOGIN_SUCCESS',
	LOGIN_FAILED = 'LOGIN_FAILED',

	LOGOUT_REQUEST = 'LOGOUT_REQUEST',
	LOGOUT_SUCCESS = 'LOGOUT_SUCCESS',
	LOGOUT_FAILED = 'LOGOUT_FAILED',

	SIGNUP_REQUEST = 'SIGNUP_REQUEST',
	SIGNUP_SUCCESS = 'SIGNUP_SUCCESS',
	SIGNUP_FAILED = 'SIGNUP_FAILED',

	PASSWORD_RESET_REQUEST = 'PASSWORD_RESET_REQUEST',
	PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS',
	PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED',

	SET_LOCALE_REQUEST = 'SET_LOCALE_REQUEST',
	SET_LOCALE_FAILED = 'SET_LOCALE_FAILED',
	SET_LOCALE_SUCCESS = 'SET_LOCALE_SUCCESS',

	RESET_AUTH = 'RESET_AUTH'
}
