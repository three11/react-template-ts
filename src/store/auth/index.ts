export * from './enums';
export * from './interfaces';
export * from './reducer';
export { default as authReducer } from './reducer';
export * from './sagas';
