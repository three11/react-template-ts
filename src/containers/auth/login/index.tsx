import * as React from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Link, useNavigate } from 'react-router-dom';

import { Routes } from '@utilities';
import { AuthActionType } from '@store/enums';
import { Wrapper, LoginForm } from '@components';

export const Login: React.FunctionComponent = () => {
	const { t } = useTranslation();
	const dispatch = useDispatch();
	const navigate = useNavigate();

	return (
		<Wrapper className="o-wrapper--fancy">
			<LoginForm
				onSubmit={(payload: any): void => {
					dispatch({
						type: AuthActionType.LOGIN_REQUEST,
						payload: {
							...payload,
							redirect: (): void => navigate(Routes.BASE)
						}
					});
				}}
			>
				<p className="c-form__hint">
					{t("Don't have an account?")} <Link to={Routes.SIGNUP}>{t('Sign up')}</Link>
				</p>

				<p className="c-form__hint">
					{t('Forgot password?')} <Link to={Routes.PASSWORD_RESET}>{t('Reset password')}</Link>
				</p>
			</LoginForm>
		</Wrapper>
	);
};

export default Login;
