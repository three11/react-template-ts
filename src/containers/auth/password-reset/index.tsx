import * as React from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Link, useNavigate } from 'react-router-dom';

import { Routes } from '@utilities';
import { AuthActionType } from '@store/enums';
import { Wrapper, PasswordResetForm } from '@components';

export const PasswordReset: React.FunctionComponent = () => {
	const { t } = useTranslation();
	const dispatch = useDispatch();
	const navigate = useNavigate();

	return (
		<Wrapper className="o-wrapper--fancy">
			<PasswordResetForm
				onSubmit={(payload: any): void => {
					dispatch({
						type: AuthActionType.PASSWORD_RESET_REQUEST,
						payload: {
							...payload,
							redirect: (): void => navigate(Routes.LOGIN)
						}
					});
				}}
			>
				<p className="c-form__hint">
					{t('You can also')} <Link to={Routes.SIGNUP}>{t('Sign up')}</Link> {t('or')}{' '}
					<Link to={Routes.LOGIN}>{t('Login')}</Link>
				</p>
			</PasswordResetForm>
		</Wrapper>
	);
};

export default PasswordReset;
