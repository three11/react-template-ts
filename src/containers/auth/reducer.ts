import { TOKEN_KEY, REFRESH_TOKEN_KEY, TOKEN_THRESHOLD_KEY } from '@utilities';

import { AuthActionType } from './enums';
import { AuthState, AuthAction } from './interfaces';

export const initialState: AuthState = {
	token: localStorage.getItem(TOKEN_KEY) || '',
	locale: 'en',
	loading: false,
	threshold: Number(localStorage.getItem(TOKEN_THRESHOLD_KEY)) || 0,
	loginError: '',
	logoutError: '',
	signupError: '',
	refreshToken: localStorage.getItem(REFRESH_TOKEN_KEY) || '',
	passwordResetError: ''
};

export default (state = initialState, { type, payload }: AuthAction): AuthState => {
	switch (type) {
		case AuthActionType.LOGIN_REQUEST:
		case AuthActionType.LOGOUT_REQUEST:
		case AuthActionType.SIGNUP_REQUEST:
		case AuthActionType.SET_LOCALE_REQUEST:
		case AuthActionType.PASSWORD_RESET_REQUEST:
			return {
				...state,
				loading: true
			};
		case AuthActionType.LOGIN_FAILED:
		case AuthActionType.LOGOUT_FAILED:
		case AuthActionType.SIGNUP_FAILED:
		case AuthActionType.SET_LOCALE_FAILED:
		case AuthActionType.PASSWORD_RESET_FAILED:
			return {
				...state,
				...payload,
				loading: false
			};
		case AuthActionType.LOGIN_SUCCESS:
		case AuthActionType.LOGOUT_SUCCESS:
		case AuthActionType.SIGNUP_SUCCESS:
		case AuthActionType.SET_LOCALE_SUCCESS:
		case AuthActionType.PASSWORD_RESET_SUCCESS:
			return {
				...state,
				...payload,
				loading: false
			};
		case AuthActionType.RESET_AUTH:
			return initialState;
		default:
			return state;
	}
};
