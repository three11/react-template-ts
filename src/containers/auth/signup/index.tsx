import * as React from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Link, useNavigate } from 'react-router-dom';

import { Routes } from '@utilities';
import { AuthActionType } from '@store/enums';
import { Wrapper, SignupForm } from '@components';

export const Signup: React.FunctionComponent = () => {
	const { t } = useTranslation();
	const dispatch = useDispatch();
	const navigate = useNavigate();

	return (
		<Wrapper className="o-wrapper--fancy">
			<SignupForm
				onSubmit={(payload: any): void => {
					dispatch({
						type: AuthActionType.SIGNUP_REQUEST,
						payload: {
							...payload,
							redirect: (): void => navigate(Routes.LOGIN)
						}
					});
				}}
			>
				<p className="c-form__hint">
					{t('Already have an account?')} <Link to={Routes.LOGIN}>{t('Login')}</Link>
				</p>
			</SignupForm>
		</Wrapper>
	);
};

export default Signup;
