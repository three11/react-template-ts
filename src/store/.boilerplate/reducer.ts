import { ExampleActionType } from './enums';
import { ExampleState, ExampleAction } from './interfaces';

export const initialState: ExampleState = {
	error: false,
	loading: true
};

export default (state = initialState, { type, payload }: ExampleAction): ExampleState => {
	switch (type) {
		case ExampleActionType.REQUEST:
			return {
				...state,
				error: false,
				loading: true
			};
		case ExampleActionType.FAILED:
			return {
				...state,
				error: true,
				loading: false
			};
		case ExampleActionType.SUCCESS:
			return {
				...state,
				error: false,
				loading: false
			};

		default:
			return state;
	}
};
