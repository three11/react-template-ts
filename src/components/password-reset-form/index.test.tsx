import * as React from 'react';
import { render } from '@testing-library/react';

import { initialState } from '@store/branches/auth/reducer';
import { PasswordResetForm } from '.';
import { TestStoreProvider } from '@components';

test('PasswordResetForm component should render successfully', () => {
	const { asFragment } = render(
		<TestStoreProvider state={{ auth: { ...initialState } }}>
			<PasswordResetForm onSubmit={(): jest.Mock => jest.fn()} />
		</TestStoreProvider>
	);

	expect(asFragment()).toMatchSnapshot();
});
