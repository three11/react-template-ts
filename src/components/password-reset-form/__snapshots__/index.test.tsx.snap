// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordResetForm component should render successfully 1`] = `
<form
  className="c-form c-form--login"
  onSubmit={[Function]}
>
  <h2>
    Reset your account's password
  </h2>
  <Field
    label="Email Address"
    name="email"
    placeholder="someone@example.com"
    register={
      Object {
        "name": "email",
      }
    }
    type="email"
  />
  <Field
    label="New password"
    name="password"
    placeholder="********"
    register={
      Object {
        "name": "password",
      }
    }
    type="password"
  />
  <Field
    label="Confirm new password"
    name="confirm"
    placeholder="********"
    register={
      Object {
        "name": "confirm",
      }
    }
    type="password"
  />
  <Button
    className=""
    disabled={false}
    type="submit"
  >
    Reset password
  </Button>
</form>
`;
